### PR TITLE
[GLUTEN-1794][VL] support split preload

### PR DESCRIPTION
**Pull request by zhli1142015**
support split preload

Fixes: #1794

### CODE REVIEW

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala [hunk 1]
```diff
@@ -756,6 +766,20 @@ object GlutenConfig {
       .booleanConf
       .createWithDefault(false)
 
+  val COLUMNAR_VELOX_CONNECTOR_IO_THREADS =
+    buildConf("spark.gluten.sql.columnar.backend.velox.IOThreads")
+      .internal()
+      .doc("The IO threads for cpnnector split preloading")
```
Comment (zhouyuan) [inline shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:772]:
```suggestion
      .doc("The IO threads for connector split preloading")
```

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala [hunk 2]
```diff
@@ -756,6 +766,20 @@ object GlutenConfig {
...
+      .intConf
+      .createWithDefault(Runtime.getRuntime.availableProcessors)
```
Comment (zhouyuan) [inline shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:774]: using all threads by default maybe too aggressive, how about set this to 1?

File: cpp/velox/compute/VeloxInitializer.cc [hunk 3]
```diff
@@ -242,7 +238,27 @@ void VeloxInitializer::initCache(const std::unordered_map<std::string, std::stri
...
+  }
+  int32_t splitPreloadPerDriver = std::stoi(kVeloxSplitPreloadPerDriverDefault);
+  got = conf.find(kVeloxSplitPreloadPerDriver);
+  if (got != conf.end()) {
+    splitPreloadPerDriver = std::stoi(got->second);
+  }
+  if (ioThreads > 0) {
+    ioExecutor_ = std::make_unique<folly::IOThreadPoolExecutor>(ioThreads);
+  }
+  FLAGS_split_preload_per_driver = splitPreloadPerDriver;
```
Comment (zhouyuan) [inline cpp/velox/compute/VeloxInitializer.cc:259]: move to L259 to `if { }` ?

**Review by zhouyuan — COMMENTED** [review]
works for me, many thanks for improving this!

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala (hunk 1 above)
Comment (zhli1142015) [reply shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala]: Thanks, updated.

File: cpp/velox/compute/VeloxInitializer.cc (hunk 3 above)
Comment (zhli1142015) [reply cpp/velox/compute/VeloxInitializer.cc]: Moved line 259 into if block.

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala (hunk 2 above)
Comment (zhli1142015) [reply shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala]: synced with yuan offline, set it as 0 to disable this feature by default.

**Review by zhouyuan — APPROVED** [review]
👍

**Pull request merged**